File: 1_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
const hash = new Map();
for(let i =0; i < nums.length; i++){
    if(hash.has(target - nums[i])){
        return [hash.get(target-nums[i]), i ]
    }
    hash.set(nums[i], i)
}
};

File: 1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(len(nums)):
            needed = target - nums[i]
            if needed in seen:
                return [seen[needed], i]
            seen[nums[i]] = i

solution = Solution()
solution.twoSum([2,7,11,15], 9 )